### DISCUSSION TITLE
Request: Ability to set a maximum video resolution

### DISCUSSION THREAD

**Discussion by ghmagnet** [aclap-dev/video-downloadhelper#398] [Help and Q&A]
### Your question

First post has to be a huge thank you for this application!

I have a number of scripts that make use of the application but they cannot discriminate on the video format resolution presented to them, so by default, the first video option listed is the one selected to be downloaded. If this first video is anything above 1920x1080 then the file size tends to be too large and this system cannot play the file without issues - just too high a data rate.

My request is can the Video Qualities settings have an option to set the maximum video format to 1920x1080 so anything above this size is no longer showing up in the top position as a selectable download option?

I have tried setting Preferred video quality to 1080p thinking it might cap the video size but the larger files (3840x2160) still show as a download option.

Regards
magnet.

### Addon & CoApp version and details

version: 8.2.2.8
channel: stable
build date: 2024-03-18
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-GB
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/home/viewer/.local/share/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/viewer",
  "converterBinary": "/home/viewer/.local/share/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"
}

**Comment by jcfrog** [comment]
Hi, 
Do you mean the "Size threshold" parameter is not working?

**Reply by ghmagnet** [reply]
There is no explaination of the units "Size threshold" or "Minimum size" refer to.

Default settings are:
Size threshold 2097152 Force detect media above this size.
Minimum size 8192 Ignore hits below this size.

Changing either of these values up or down seems to make no difference to the issue of larger resolution files being shown in the download menu.

Looking at the file details shows the masterManifest listing all the video resolutions available:
...multi=256x144:144p:,426x240:240p:,854x480:480p:,1280x720:720p:,1920x1080:1080p:,3840x2160:2160p:...

so just need a setting that allows the user to set the highest resolution that the download selection menu will list and not show that 3840x2160:2160p file.
